ncm-network/src/main/perl/network.pm

```diff
@@ -809,20 +933,49 @@ sub Configure
     } else {
         foreach my $if (sort keys %ifdown) {
             # how do we actually know that the device was up?
-            # eg for non-existing device eth4: /sbin/ifdown eth4 --> usage: ifdown <device name>
+            # Do not try to down non-existing devices
+            next if (-d "/sys/class/net" && ! -e "/sys/class/net/$if");
             push(@cmds, ["/sbin/ifdown", $if]);
         }
     }
     $self->runrun(@cmds);
     # replace modified/new files
+    my $was_change = 0;
     foreach my $file (keys %exifiles) {
         if (($exifiles{$file} == 1) || ($exifiles{$file} == 2)) {
             copy($self->gen_backup_filename($file).FAILED_SUFFIX,$file) ||
                 $self->error("replace modified/new files: can't copy ",
                              $self->gen_backup_filename($file).FAILED_SUFFIX,
                              " to $file. ($!)");
+            $was_change = 1;
         } elsif ($exifiles{$file} == -1) {
             unlink($file) || $self->error("replace modified/new files: can't unlink $file. ($!)");
+            $was_change = 1;
+        }
+    }
+    ## kick udev to rename devices if needed; do nothing if no files were changed
+    if ($was_change) {
+        # Remove the auto-generated rule file if present
+        if ($config->elementExists("/software/components/network/udev_rules_file")) {
```

Comment:
metaconfig has udev related TT files, maybe we can generate the udev files with metaconfig, and call the trigger/settle here?

Reply:
I don't want to _create_ any udev files - but I need to _remove_ the one udev generates automatically. If the auto-generated file exists, then udev will not look into the ifcfg-* files, and it will not notice that some interfaces which did not have a configuration before may need to be renamed.

Reply:
what `autogenerates` it? anaconda? i always tought that ifcfg file preceded udev rules if HWADDR was set.
the location of this autogenerated udev file is OS release specific i guess? it would be nice to add the value somewhere (for aii we have os flavours); or do you think we can hardcode the location?